aiohttp/connector.py

```diff
@@ -392,11 +396,18 @@ def closed(self):
 
             try:
                 await fut
-            finally:
-                # remove a waiter even if it was cancelled
-                waiters.remove(fut)
```

Comment:
this is slow because it's trying to find a future in a list for each request that's waiting on a connector